### PR TITLE
fix(WebWorkers): Add support for transitionend events.

### LINKED ISSUE #6649: transitionend not supported on WebWorkers
We need to include a serializer for the transitionend event.

### PR DESCRIPTION
**Pull request by jteplitz**
Closes #6649

### CODE REVIEW

File: modules/angular2/src/web_workers/ui/event_serializer.dart [hunk 1]
```diff
@@ -86,6 +86,14 @@ Map<String, dynamic> serializeKeyboardEvent(dynamic e) {
   return serialized;
 }
 
+Map<String, dynamic> serializeTransitionEvent(dynamic e) {
+  var serialized = serializeGenericEvent(e);
+  serialized['propertyName'] = e.propertyName;
+  serialized['elapsedTime'] = e.propertyName;
```
Comment (jeffbcross) [inline modules/angular2/src/web_workers/ui/event_serializer.dart]: Why not `e.elapsedTime`? Same question for next line.

Comment (jteplitz602) [inline modules/angular2/src/web_workers/ui/event_serializer.dart]: Good catch. Fixed.

File: modules/angular2/src/web_workers/ui/event_serializer.dart [hunk 2]
```diff
@@ -86,6 +86,14 @@ Map<String, dynamic> serializeKeyboardEvent(dynamic e) {
   return serialized;
 }
 
+Map<String, dynamic> serializeTransitionEvent(dynamic e) {
+  var serialized = serializeGenericEvent(e);
+  serialized['propertyName'] = e.propertyName;
+  serialized['elapsedTime'] = e.elapsedTime;
+  serialized['psuedoElement'] = e.psuedoElement;
```
Comment (jeffbcross) [inline modules/angular2/src/web_workers/ui/event_serializer.dart]: Mis-spelled :)

`pseudoelement`

Comment (jteplitz602) [inline modules/angular2/src/web_workers/ui/event_serializer.dart]: That's awkward. Fixed.

File: modules/angular2/src/web_workers/worker/event_deserializer.dart [hunk 3]
```diff
@@ -31,6 +31,10 @@ class GenericEvent {
   Point get page => _getPoint('page');
   Point get screen => _getPoint('screen');
 
+  String get propertyName => properties['propertyName'];
+  num get elapsedTime => properties['elapsedTime'];
+  String get psuedoElement => properties['psuedoElement'];
```
Comment (campersau) [inline modules/angular2/src/web_workers/worker/event_deserializer.dart]: This is also mis-spelled.
`pseudoelement`

Comment (jteplitz602) [inline modules/angular2/src/web_workers/worker/event_deserializer.dart]: Thanks!

**Comment by rkirov** [comment]
flaky saucelabs is failing https://travis-ci.org/angular/angular/builds/105066438

manually pushed at https://github.com/angular/angular/commit/c2a38c05aa898a872b82e57a2183bcd2d47c9da9

**Pull request closed without merging**